homeassistant/components/lifx/coordinator.py

```diff
@@ -225,6 +226,36 @@ async def _async_update_data(self) -> None:
             if lifx_features(self.device)["infrared"]:
                 await async_execute_lifx(self.device.get_infrared)
 
+        except asyncio.TimeoutError as ex:
+            if self._timeouts == 0:
+                self._offline_time = monotonic()
+
+            self._timeouts += 1
+
+            if self._timeouts >= MAX_TIMEOUTS_TO_DECLARE_UPDATE_FAILED:
+                raise UpdateFailed(
+                    f"The device failed to respond after {MAX_TIMEOUTS_TO_DECLARE_UPDATE_FAILED} attempts"
+                ) from ex
+
+            _LOGGER.debug(
+                "Incrementing timeout counter to %s after no reply from %s (%s)",
+                self._timeouts,
+                self.device.label,
+                self.device.ip_addr,
+            )
+
+            await self._async_update_data()
```

Comment:
I'm not following why an update is being dispatched to write state here. It seems like nothing has changed since we timed out?

Reply:
I'm not writing state, I'm restarting the same method so that the update isn't (yet) marked as failed by Home Assistant.

Reply:
Sorry. I misread it the first time and adjusted the comment